ebos/ecloutputblackoilmodule.hh

```diff
@@ -1311,45 +1339,41 @@ public:
                 
                                 const auto flowtype = ftype(injType);
                                 const auto flowctl = fctl(ctlMode);
+                                Scalar flowPot = 0.0;
                                 if(flowtype == "Oil") //WellInjDataType::CTRLModeOil
                                 {                    
                                         if (flowctl == "RATE"){ tmp_names[1] = "ORAT"; }
                                         else { tmp_names[1] =  flowctl; }
+                                        flowPot = tmp_values[2];
                                 } 
                                 else if (flowtype == "Wat") //WellInjDataType::CTRLModeWat
                                 {
                                         if (flowctl == "RATE"){ tmp_names[3] = "WRAT"; }
                                         else { tmp_names[2] =  flowctl; }
+                                        flowPot = tmp_values[3];
                                 } 
                                 else if (flowtype == "Gas") //WellInjDataType::CTRLModeGas
                                 {
                                         if (flowctl == "RATE"){ tmp_names[3] = "GRAT"; }
                                         else { tmp_names[3] =  flowctl; }
-                                }                                       
-
-                                tmp_values[0] = well.getHeadI() + 1; //WellInjDataType::wellLocationi
-                                tmp_values[1] = well.getHeadJ() + 1; //WellInjDataType::wellLocationj 
-                                tmp_values[2] = get("WOIR"); //WellInjDataType::OilRate
-                                tmp_values[3] = get("WWIR"); //WellInjDataType::WaterRate
-                                tmp_values[4] = get("WGIR"); //WellInjDataType::GasRate 
-                                tmp_values[5] = get("WVIR");//WellInjDataType::FluidResVol                              
-                                tmp_values[6] = get("WBHP"); //WellInjDataType::BHP
-                                tmp_values[7] = get("WTHP"); //WellInjDataType::THP
-                                //tmp_values[8] = 0; //WellInjDataType::SteadyStateII
+                                        flowPot = tmp_values[4];                                        
+                                }                                  
+                                
+                                tmp_values[8] = flowPot/(tmp_values[6] - reservoirAveragePressure_); //WellInjDataType::SteadyStateII
```

Comment:
I assume this is the correct formula for the (prod/inj)ectivity index, but I am a little puzzled by the average pressure being involved?

Reply:
I was/am not sure about what to output.; this eq. is what came out from discussions during the meeting. Should I get the well index and well potential directly from well state?